### PR TITLE
samples: fs: fat_fs: support on stm32h747i-disco  board

### PR DESCRIPTION
**Pull request by mkaricheri**
This patch allow demonstrating fat_fs sample application on stm32h747i-disco  board with SD card.

### CODE REVIEW

**Comment by Laczen** [comment]
@mkaricheri, the use of "fixed partitions" on a disk is not a good idea. When used in combination with flash_map there will be a possibility to use flash_area routines on that partition, which will not work. At the moment there is no possibility to define partitions on disks,  so it is also not possible to use littlefs/fatfs on a disk partition.

**Comment by mkaricheri** [comment]
> @mkaricheri, the use of "fixed partitions" on a disk is not a good idea. When used in combination with flash_map there will be a possibility to use flash_area routines on that partition, which will not work. At the moment there is no possibility to define partitions on disks, so it is also not possible to use littlefs/fatfs on a disk partition.

You are right. I have removed the fixed partition. However it needs an overlay to add a zephyr,sdmmc-disk under the sdmmc1
node in the board dts to have a disk driver enabled. After the the sample would run on this board and I can see the FAT FS file system running on my board. The update is pushed.

But for LittleFS I find something wrong in your comment. I agree that flash_map is not available for disk. However the littleFS does work on my board over disk on my board, stm32H747i-disco and some other STM boards currently supported. It doesn't use the fstab parameters or the default config variables for the same and the same is read from the hardware such as Block size and set the parameters based on that  https://github.com/zephyrproject-rtos/zephyr/blob/3dce4c7ba39fc7e436aee99e6d5dada70380f68f/subsys/fs/littlefs_fs.c#L846

So I have removed the same from the DTS for littleFS and updated this PR.

**Comment by mkaricheri** [comment]
Here is the log for FAT FS on STM32H747i-disco

*** Booting Zephyr OS build zephyr-v3.5.0-1023-g36d5e4833855 ***
[00:00:00.020,000] <inf> main: Block count 31116288
Sector size 512
Memory Size(MB) 15193
Disk mounted.

Listing dir /SD: ...
[FILE] SOME.DAT (size = 0)
[DIR ] SOME

This is the log for LittleFS
*** Booting Zephyr OS build zephyr-v3.5.0-1023-gf911accc19b3 ***
Sample program to r/w files on littlefs
I: LittleFS version 2.5, disk version 2.0
I: FS at SDMMC: is 31116288 0x200-byte blocks with 512 cycle
I: sizes: rd 512 ; pr 512 ; ca 512 ; la 2048
E: WEST_TOPDIR/modules/fs/littlefs/lfs.c:1234: Corrupted dir pair at {0x0, 0x1}
W: can't mount (LFS -84); formatting
I: /SDMMC: mounted
/SDMMC:: bsize = 512 ; frsize = 512 ; blocks = 31116288 ; bfree = 31116286

Listing dir /SDMMC: ...
/SDMMC:/boot_count read count:0 (bytes: 0)
/SDMMC:/boot_count write new boot count 1: [wr:1]
I: Test file: /SDMMC:/pattern.bin not found, create one!
------ FILE: /SDMMC:/pattern.bin ------
01 55 55 55 55 55 55 55 02 55 55 55 55 55 55 55
03 55 55 55 55 55 55 55 04 55 55 55 55 55 55 55
05 55 55 55 55 55 55 55 06 55 55 55 55 55 55 55
07 55 55 55 55 55 55 55 08 55 55 55 55 55 55 55
09 55 55 55 55 55 55 55 0a 55 55 55 55 55 55 55
0b 55 55 55 55 55 55 55 0c 55 55 55 55 55 55 55
0d 55 55 55 55 55 55 55 0e 55 55 55 55 55 55 55
0f 55 55 55 55 55 55 55 10 55 55 55 55 55 55 55
11 55 55 55 55 55 55 55 12 55 55 55 55 55 55 55
13 55 55 55 55 55 55 55 14 55 55 55 55 55 55 55
15 55 55 55 55 55 55 55 16 55 55 55 55 55 55 55
17 55 55 55 55 55 55 55 18 55 55 55 55 55 55 55
19 55 55 55 55 55 55 55 1a 55 55 55 55 55 55 55
1b 55 55 55 55 55 55 55 1c 55 55 55 55 55 55 55
1d 55 55 55 55 55 55 55 1e 55 55 55 55 55 55 55
1f 55 55 55 55 55 55 55 20 55 55 55 55 55 55 55
21 55 55 55 55 55 55 55 22 55 55 55 55 55 55 55
23 55 55 55 55 55 55 55 24 55 55 55 55 55 55 55
25 55 55 55 55 55 55 55 26 55 55 55 55 55 55 55
27 55 55 55 55 55 55 55 28 55 55 55 55 55 55 55
29 55 55 55 55 55 55 55 2a 55 55 55 55 55 55 55
2b 55 55 55 55 55 55 55 2c 55 55 55 55 55 55 55
2d 55 55 55 55 55 55 55 2e 55 55 55 55 55 55 55
2f 55 55 55 55 55 55 55 30 55 55 55 55 55 55 55
31 55 55 55 55 55 55 55 32 55 55 55 55 55 55 55
33 55 55 55 55 55 55 55 34 55 55 55 55 55 55 55
35 55 55 55 55 55 55 55 36 55 55 55 55 55 55 55
37 55 55 55 55 55 55 55 38 55 55 55 55 55 55 55
39 55 55 55 55 55 55 55 3a 55 55 55 55 55 55 55
3b 55 55 55 55 55 55 55 3c 55 55 55 55 55 55 55
3d 55 55 55 55 55 55 55 3e 55 55 55 55 55 55 55
3f 55 55 55 55 55 55 55 40 55 55 55 55 55 55 55

41 55 55 55 55 55 55 55 42 55 55 55 55 55 55 55
43 55 55 55 55 55 55 55 44 55 55 55 55 55 55 55
45 55 aa
I: /SDMMC: unmounted
/SDMMC: unmount: 0

**Comment by Laczen** [comment]
@mkaricheri, fatfs and littlefs can be used on entire disks, but as disk partitions do not exist in zephyr they cannot be used on part of a disk. The setup of littlefs trough fstab should not be used for disks (unless of course the partition property in fstab is changed into something that would support disks). I have seen you made a PR for littlefs but had no time to review (was on holiday). I would have made the same remark, but it was already merged.

**Comment by mkaricheri** [comment]
@erwango I realized that fstab entries for STM32H747i-disco is not used for LittleFS. So I have removed the same in this PR. Please review and comment

**Comment by Laczen** [comment]
@mkaricheri, my remark was only regarding the /delete-node line.

Please make the correction for littlefs in a separate PR.

**Comment by mkaricheri** [comment]
> @mkaricheri, my remark was only regarding the /delete-node line.
>
OK 
> Please make the correction for littlefs in a separate PR.

Sure thing!..

**Comment by mkaricheri** [comment]
> > @mkaricheri, my remark was only regarding the /delete-node line.
> 
> OK
> 
> > Please make the correction for littlefs in a separate PR.
> 
> Sure thing!..

@Laczen I have updated the commit. Could you also check if my sample.yaml update is correct. 

> > @mkaricheri, my remark was only regarding the /delete-node line.
> 
> OK
> 
> > Please make the correction for littlefs in a separate PR.
> 
> Sure thing!..

@Laczen removed LittleFS changes. removed delete-node and pushed the update. Could you please check if my sample.yaml update is correct. I just followed the existing yaml content and may not be correct.

**Comment by mkaricheri** [comment]
> @erwango I realized that fstab entries for STM32H747i-disco is not used for LittleFS. So I have removed the same in this PR. Please review and comment

Added another PR to fix this. Fyi @erwango

**Review by Laczen — DISMISSED** [review]
LGTM, thanks.

**Comment by mkaricheri** [comment]
> LGTM, thanks.

Thanks @Laczen for the approval. @nashif, @de-nordic if one of you can review and approve, that will help to merge this. Thanks!

File: samples/subsys/fs/fs_sample/sample.yaml [hunk 1]
```diff
@@ -45,3 +45,9 @@ tests:
   sample.filesystem.ext2:
     extra_args: CONF_FILE="prj_ext.conf"
     platform_allow: hifive_unmatched bl5340_dvk_cpuapp
+  sample.filesystem.fat_fs.stm32h747i_disco_m7_sdmmc:
+    build_only: true
+    filter: dt_compat_enabled("zephyr,sdmmc-disk")
```
Comment (erwango) [inline samples/subsys/fs/fs_sample/sample.yaml:50]: Filter is superfluous

**Review by erwango — CHANGES_REQUESTED** [review]

File: samples/subsys/fs/fs_sample/sample.yaml (hunk 1 above)
Comment (mkaricheri) [reply samples/subsys/fs/fs_sample/sample.yaml]: OK removed and pushed

**Review by erwango — APPROVED** [review]

**Review by Laczen — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**